### DISCUSSION TITLE
Live Preview with preview-tui

### DISCUSSION THREAD

**Discussion by levone1** [jarun/nnn#1677] [General]
Live previews working great with preview-tabbed plugin, but when using preview-tui, only thumbnails, (for videos, audio files, gifs, etc.).  Is this just the way it is, or am I missing something to get the live previews to be live with preview-tui?

Thanks

edit - just discovered gifs actually are live in ...-tui, but videos just show thumbnail, and audio files just show file info...

**Comment by luukvbaal** [comment]
That is to be expected. A terminal cannot play video or audio...

**Reply by levone1** [reply]
Thanks...

Not that I would probably get it, but what's the dumbed-down explanation - if a terminal cannot play video or audio, what is playing video and audio in the Xterm window that opens when using preview-tabbed, or in other terminal preview programs that do it?

Just curious.

**Reply by luukvbaal** [reply]
tabbed does not open an xterm window, it opens a regular X window and utilizes the XEmbed protocol to embed your media viewers, all regular X11 programs. If other terminal preview programs play video or audio "in" a terminal, they use a terminal emulator specific protocol that is not supported by all terminals, and not supported by `preview-tui` currently (not necessarily impossible to support although I'm not aware of any such protocol).

**Reply by N-R-K** [reply]
> A terminal cannot play video

mpv can play videos on terminals that support true color with `--vo=tct`. But the quality is, as you'd expect, fairly bad.

**Reply by levone1** [reply]
Thanks.  The only program I can vouch for first-hand is an abandoned Ranger-type project called Hunter.  It's written in Rust, and uses it's own media player, (hunter-media), to play live previews.  How it does it technically, is over my head, but I'm sure it's as you say.

I still have my hunter bins that I built with cargo over a year ago, and can't build it anymore since there's been no Hunter updates for 2 or 3 years, and Rust won't support building it as-is anymore, (as far as I can tell, [I did try with several different toolchains, and I already had to use an old toolchain to build it before, but now even those won't work]), but still functions for now...
